extensions/control-plane/api/data-management/catalog-api/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/catalog/model/CatalogRequestDto.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.catalog.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.PositiveOrZero;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.SortOrder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@JsonDeserialize(builder = CatalogRequestDto.Builder.class)
+public class CatalogRequestDto {
+    @PositiveOrZero(message = "offset must be greater or equal to zero")
+    private Integer offset = 0;
+    @Positive(message = "limit must be greater than 0")
+    private Integer limit = 50;
+    private SortOrder sortOrder = SortOrder.ASC;
+    private List<Criterion> filter = new ArrayList<>();
```

Comment:
Better use the `CriterionDto`, currently it's used in the `contractdefinition-api` module, but I would move it upstream to `api-core`, that will permit criteria validation.

Reply:
ah good catch. Forgot we had that :)